client/my-sites/checkout/checkout-thank-you/domain-mapping-details.jsx

```diff
@@ -10,128 +10,52 @@ import { localize } from 'i18n-calypso';
-						"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-						'(where you purchased the domain), and update your name servers to:',
-					{
-						components: {
-							registrarSupportLink: registrarSupportLink,
-						},
-					}
-				) }
-			</p>
-			<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-				<li>ns1.wordpress.com</li>
-				<li>ns2.wordpress.com</li>
-				<li>ns3.wordpress.com</li>
-			</ul>
-		</div>
-	);
-
-	if ( isSubdomainMapping ) {
-		instructions = (
-			<div>
-				<p>
-					{ translate(
-						'To point your domain at your WordPress.com site, log in to your ' +
-							"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-							'(where you purchased the domain), and edit the DNS records to add a CNAME record:',
-						{
-							components: {
-								registrarSupportLink: registrarSupportLink,
-							},
-						}
-					) }
-				</p>
-				<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-					<li>
-						{ domain }. IN CNAME { selectedSiteDomain }.
-					</li>
-				</ul>
-			</div>
-		);
-	}
-
-	if ( isSubdomainMapping && isBusinessPlan ) {
-		instructions = (
-			<div>
-				<p>
-					{ translate(
-						'To point your domain at your WordPress.com site, log in to your ' +
-							"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-							'(where you purchased the domain), and edit the DNS records to add these NS records:',
-						{
-							components: {
-								registrarSupportLink: registrarSupportLink,
-							},
-						}
-					) }
-				</p>
-				<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-					<li>{ domain }. IN NS ns1.wordpress.com.</li>
-					<li>{ domain }. IN NS ns2.wordpress.com.</li>
-					<li>{ domain }. IN NS ns3.wordpress.com.</li>
-				</ul>
-			</div>
-		);
-	}
-
-	const description = (
-		<div>
-			{ instructions }
-			<p>{ translate( 'If you already did this, no further action is required.' ) }</p>
-		</div>
+	const mappingInstructions = (
+		<DomainMappingInstructions
+			aRecordsRequiredForMapping={ purchasedDomain?.aRecordsRequiredForMapping }
+			areDomainDetailsLoaded={ ! isRequestingDomainsDetails && !! purchasedDomain }
+			domainName={ purchasedDomain?.name }
+		/>
 	);
 
 	return (
 		<div className="checkout-thank-you__domain-mapping-details">
-			<PurchaseDetail
-				icon="cog"
-				description={ description }
-				buttonText={ translate( 'Learn more' ) }
-				href={ isSubdomainMapping ? MAP_SUBDOMAIN : MAP_EXISTING_DOMAIN }
-				target="_blank"
-				rel="noopener noreferrer"
-				isRequired
-			/>
+			<QuerySiteDomains siteId={ siteId } />
+			<PurchaseDetail icon="cog" description={ mappingInstructions } isRequired />
 		</div>
 	);
 };
 
 const mapStateToProps = ( state, { domain } ) => {
 	const selectedSite = getSelectedSite( state );
 	return {
+		domains: getDomainsBySiteId( state, selectedSite.ID ),
```

Comment:
```suggestion
	const selectedSite = getSelectedSite( state );
	const domains = getDomainsBySiteId( state, selectedSite.ID );
	return {
		purchasedDomain: getSelectedDomain( { domains, selectedDomainName: domain } );
```

I'd move fetching the selected domain name to here, instead of doing it inside the component.